### ISSUE TITLE
[Bug]: if a control option includes 'reverse' , `function reverse() { [native code] }` is displayed

### ISSUE THREAD

**Issue by jericirenej** [storybookjs/storybook#30142]
### Describe the bug

If a story control type contains the word 'reverse' among its options, the control value in the UI for this option will display the text ` `function reverse() { [native code] }`  instead of "reverse".

I've provided the StackBlitz reproduction link below, however it is from an anonymous fork, so it might get wiped after a while. In that case, the story setup is quite straightforward:

```html
<script module>
  import { defineMeta } from '@storybook/addon-svelte-csf';
  import SomeComponentComponent from './SomeComponent.svelte';

  const { Story } = defineMeta({
    title: 'Example/SomeComponent',
    component: SomeComponentComponent,
    tags: ['autodocs'],
    argTypes: {
      option: {
        control: "inline-radio",
        options: ['normal', 'reverse',],
      },
    },
    
  });
</script>

<Story name="SomeComponent" args={{ option:'normal' }} />
```

### Reproduction link

https://stackblitz.com/edit/github-yndwrtai?file=src%2Fstories%2FSomeComponent.stories.svelte

### Reproduction steps

1. Create a component that accepts a string output
2. In the component's story, specify the argType as `{control: "inline-radio", options: ["normal", "reverse"]}`
3. Run Storybook and navigate to the component story. For the specified control, you will see the following options displayed:
```
• normal 
• function reverse() { [native code] }
```

### System

```bash
torybook Environment Info:

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm <----- active
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @storybook/addon-essentials: ^8.5.0-beta.6 => 8.5.0-beta.6 
    @storybook/addon-interactions: ^8.5.0-beta.6 => 8.5.0-beta.6 
    @storybook/addon-svelte-csf: ^5.0.0-next.21 => 5.0.0-next.21 
    @storybook/blocks: ^8.5.0-beta.6 => 8.5.0-beta.6 
    @storybook/svelte: ^8.5.0-beta.6 => 8.5.0-beta.6 
    @storybook/svelte-vite: ^8.5.0-beta.6 => 8.5.0-beta.6 
    @storybook/test: ^8.5.0-beta.6 => 8.5.0-beta.6 
    storybook: ^8.5.0-beta.6 => 8.5.0-beta.6
```

**Comment by grantralls** [comment]
I'm going to take a crack at this unless anyone says otherwise. It seems the root cause is the `normalizeOptions` function [here](https://github.com/storybookjs/storybook/blob/next/code/addons/docs/src/blocks/controls/options/Options.tsx#L8-L29). This bug occurs when the `labels` param is an array and `item` on line 24 is "reverse". Which resolves the option to an array function. I'm going to figure out _why_ labels is an array because it should be an object given the `Record` type. I have verified the labels prop is an array through react devtools.

**Comment by grantralls** [comment]
A fix that "works" is to change `control` to an object an manually assign labels. So the new story would look like this.

```svelte
<script module>
  import { defineMeta } from '@storybook/addon-svelte-csf';
  import SomeComponentComponent from './SomeComponent.svelte';

  const { Story } = defineMeta({
    title: 'Example/SomeComponent',
    component: SomeComponentComponent,
    tags: ['autodocs'],
    argTypes: {
      option: {
        control: { type: "inline-radio", labels: {"normal": "normal", "reverse": "reverse"}},
        options: ['normal', 'reverse',]
      },
    },
    
  });
</script>

<Story name="SomeComponent" args={{ option:'normal' }} />
```

Obviously not ideal. I'm still diving into where those labels are coming from. ~~Interestingly enough, I'm unable to reproduce with svelte-vite-default-ts sandbox.~~ I'm going to use my own repo where I can reproduce, and link it as a sandbox and go down that path to see if I find something.

**Comment by grantralls** [comment]
I wasn't able to use my local project as a template as the templating setup uses yarn and my project does not. What I was able to find is the StorybookAPI. It is a context that within it, has a key-value map of a story name with the value being an object containing a lot of data on the individual story's capabilities. Within there is the labels data. For my local project the labels field is present as an array. When in reality it should not be present at all since I don't provide labels in my defineMeta call. Much less an array. My fix above still works, it just isn't pretty.

I believe I'm reaching the point in which I'm in over my head. I'm going to take one more crack at finding out where that StorybookAPI data gets set. And how it could be an array. I'm starting to suspect it could have something to do with specific versions of storybook. I tried to get a debugger up and running but since storybook's files have a build step, that build config is hidden in prep.ts, it's non-trivial to generate debug sourcemaps and get it working. It would be really nice if that was just part of the repo or the contributing docs.

**Comment by grantralls** [comment]
I don't think it has anything to do with the `defineMeta` call and everything to do with type inference. When a component prop has a union type, the labels get generated as an array. The following type would create the labels array within the StorybookAPI. I was able to reproduce on the storybook svelte-vite-default-ts template but not the react one.

```ts
type ComponentProps = {
    size: 'small' | 'medium' | 'large';
};
```

**Comment by grantralls** [comment]
Here it is!

https://github.com/storybookjs/storybook/blob/next/code/renderers/svelte/src/extractArgTypes.ts#L111-L126

Now to figure out _why_ that was ever done and should it be changed.

**Comment by Sidnioulz** [comment]
Fixed by https://github.com/storybookjs/storybook/pull/31980.

**Comment by jericirenej** [comment]
Great to see this resolved, thanks @grantralls!
